src/calendarworker.cpp

```diff
@@ -584,8 +584,13 @@ CalendarWorker::eventOccurrences(const QList<CalendarData::Range> &ranges) const
     const QStringList excluded = excludedNotebooks();
     QHash<QString, CalendarData::EventOccurrence> filtered;
     for (const CalendarData::Range range : ranges) {
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+        KCalendarCore::OccurrenceIterator it(*mCalendar, QDateTime(range.first.addDays(-1).startOfDay()),
```

Comment:
I think you can remove the `QDateTime()` here and in the next line since `.startOfDay()` and `.endOfDay()` is already returning a QDateTime object.